include/ametsuchi/buffer.h

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <ametsuchi/exception.h>
+#include <ametsuchi/globals.h>
+#include <cmath>
+
+namespace ametsuchi {
+namespace buffer {
+
+/**
+ * Basic circular buffer
+ */
+template <typename T>
+class CircularStack {
```

Comment:
1. Yeah, that should be fixed, right
2. Yes for following reasons:
- std::stack != circular buffer
- std::stack doesn't have random access iface
- this class should be extended in future, so it's able to contain array of that buffers